### DISCUSSION TITLE
Tools to split one boundary into two

### DISCUSSION THREAD

**Discussion by pj131611** [idaholab/moose#28642] [Q&A Meshing] [answered]
Hi there,

While using "Thermal Contact" for mutually contacting geometry, we need to find a way to turn the interface or intersection between two geometry into two different boundaries. But I'm stuck on how to do so while the two contacting objects are in different size, let's say, a cube with length of 1 and a cube with length 2. Do I have to cut the larger cube in this case?

Any help will be much appreciated!

**Comment by GiudGiud** [comment, marked as answer]
Hello

The contact interfaces do not have to match.
If your geometries are cubes, the sidesets-generators in moose (under MeshGenerator) should easily add the new sidesets
(bottom of https://mooseframework.inl.gov/syntax/Mesh/index.html)

If your mesh started out with the two cubes merged together, you can use the BreakMeshByBlockGenerator to split them
https://mooseframework.inl.gov/source/meshgenerators/BreakMeshByBlockGenerator.html
Guillaume

**Reply by pj131611** [reply]
Thanks for the quick response! That's exactly what I need! 

On top of that, some of the interface are connected while I would like to take care of them one by one. So is there a function that breaks orthogonal connected interface into single pieces?

**Reply by GiudGiud** [reply]
do you mean cutting subdomains apart? or do you mean cutting a large boundary in multiple parts?

**Reply by pj131611** [reply]
Cutting a large boundary into multiple parts.

**Reply by GiudGiud** [reply]
can you use the ParsedGenerateSideset with `include_boundaries` to do that?

**Reply by pj131611** [reply]
I always run into error while using Parsed related functions, but I manage to split the interface with SideSetsAroundSubdomainGenerator by setting different normal vectors (benefit of cubes :D), but I noticed that some of the side side doesn't have a corresponding node set and the name of side set and node set doesn't agree either, did I missed anything?

**Reply by GiudGiud** [reply]
If there is a missing nodeset you can use an option in the Mesh block to convert the sidesets to nodeset s

There is also a generator for this conversion

SidesetsToNodesetsGenerator iirc

**Reply by pj131611** [reply]
That's Wonderful. Thanks again for all your help!
